signer/tests/integration/postgres.rs

```diff
@@ -5399,3 +5399,1200 @@ async fn pending_rejected_withdrawal_already_accepted() {
+    }
+
+    /// Creates and stores a withdrawal request, confirmed in the specified
+    /// bitcoin & stacks blocks.
+    ///
+    /// If votes are provided, signer vote entries are created and stored as
+    /// well (with the same number of votes as the length of the `votes` slice).
+    /// The votes are stored under random signer public keys.
+    async fn store_withdrawal_request(
+        db: &PgStore,
+        request_id: u64,
+        bitcoin_block: &BitcoinBlock,
+        stacks_block: &StacksBlock,
+        votes: &[bool],
+    ) -> WithdrawalRequest {
+        let withdrawal_request = WithdrawalRequest {
+            request_id,
+            block_hash: stacks_block.block_hash,
+            bitcoin_block_height: bitcoin_block.block_height,
+            ..Faker.fake()
+        };
+
+        db.write_withdrawal_request(&withdrawal_request)
+            .await
+            .expect("failed to write withdrawal request");
+
+        store_votes(db, &withdrawal_request, votes).await;
+
+        withdrawal_request
+    }
+
+    /// Creates a withdrawal rejection event in the database at the specified
+    /// stacks block.
+    async fn reject_withdrawal_request(
+        db: &PgStore,
+        request: &WithdrawalRequest,
+        stacks_block: &StacksBlock,
+    ) {
+        db.write_withdrawal_reject_event(&WithdrawalRejectEvent {
+            request_id: request.request_id,
+            block_id: stacks_block.block_hash,
+            ..Faker.fake()
+        })
+        .await
+        .expect("failed to write withdrawal reject event");
+    }
+
+    /// Creates a sweep transaction that includes the specified withdrawal
+    /// request. The sweep transaction is written to the database in the
+    /// provided bitcoin block and the transaction ID is returned.
+    async fn sweep_withdrawal_request(
+        db: &PgStore,
+        request: &WithdrawalRequest,
+        at_bitcoin_block: &BitcoinBlockHash,
+    ) -> BitcoinTxId {
+        // Simulate a sweep transaction on the canonical chain which will
+        // include the withdrawal request.
+        let transaction = model::Transaction {
+            txid: Faker.fake(),
+            block_hash: at_bitcoin_block.into_bytes(),
+            tx_type: model::TransactionType::SbtcTransaction,
+            tx: Vec::new(),
+        };
+        let bitcoin_sweep_tx = model::BitcoinTxRef {
+            txid: transaction.txid.into(),
+            block_hash: *at_bitcoin_block,
+        };
+        db.write_transaction(&transaction)
+            .await
+            .expect("failed to write transaction");
+        db.write_bitcoin_transaction(&bitcoin_sweep_tx)
+            .await
+            .expect("failed to write bitcoin transaction");
+
+        // Write a fully validated withdrawal output for the request.
+        db.write_bitcoin_withdrawals_outputs(&[model::BitcoinWithdrawalOutput {
+            bitcoin_txid: bitcoin_sweep_tx.txid,
+            bitcoin_chain_tip: *at_bitcoin_block,
```

Comment:
nit: `bitcoin_chain_tip` cannot be the same as `bitcoin_sweep_tx` and `transaction` `block_hash` because of causality, but I guess we don't compare them anywhere; anyway, nothing to do, just flagging it.

Reply:
Yeah I think I was just putting values that satisfied FK's and were valid for the canonical chain without needing to create extra stuff since it wasn't needed for the validations these tests are doing; but that was days ago, don't remember exactly the thought process 😅